app/Console/Command/MoveEvents.php

```diff
@@ -0,0 +1,139 @@
+ */
+class MoveEvents extends AbstractCommand
+{
+    private const PLACE_UUID = 'place-uuid';
+    private const QUERY = 'query';
+    private const FORCE = 'force';
+    private const DRY_RUN = 'dry-run';
+
+    private ResultsGeneratorInterface $searchResultsGenerator;
+
+    public function __construct(
+        CommandBus $commandBus,
+        SearchServiceInterface $searchService
+    ) {
+        parent::__construct($commandBus);
+        $this->searchResultsGenerator = new ResultsGenerator(
+            $searchService,
+            new Sorting('created', 'asc'),
+            100
+        );
+    }
+
+    public function configure(): void
+    {
+        $this
+            ->setName('event:move')
+            ->setDescription('Move an event from one place to another based on a SAPI3 query')
+            ->addArgument(
+                self::PLACE_UUID,
+                null,
+                'Place uuid to move events to.'
+            )
+            ->addArgument(
+                self::QUERY,
+                null,
+                'SAPI3 query for which events to move.'
+            )
+            ->addOption(
+                self::FORCE,
+                null,
+                InputOption::VALUE_NONE,
+                'Skip confirmation.'
+            )
+            ->addOption(
+                self::DRY_RUN,
+                null,
+                InputOption::VALUE_NONE,
+                'Execute the script as a dry run.'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): ?int
+    {
+        $placeUuid = $input->getArgument(self::PLACE_UUID);
+        $query = $input->getArgument(self::QUERY);
+
+        if ($placeUuid === null || $query === null) {
+            $output->writeln('<error>Missing argument, the correct syntax is: event:move "place_uuid_to_move_to" "sapi3_query"</error>');
+            return self::FAILURE;
+        }
+
+        $count = $this->searchResultsGenerator->count($query);
+
+        if ($count <= 0) {
+            $output->writeln('<error>No events found</error>');
+            return self::FAILURE;
+        }
+
+        if (!$this->askConfirmation($input, $output, $count)) {
+            return self::FAILURE;
+        }
+
+        $progressBar = new ProgressBar($output, $count);
+
+        $exceptions = [];
+        foreach ($this->searchResultsGenerator->search($query) as $event) {
+            try {
+                $command = new UpdateLocation($event->getId(), new LocationId($placeUuid));
+                if (!$input->getOption(self::DRY_RUN)) {
+                    $this->commandBus->dispatch($command);
+                }
+                $output->writeln('Dispatching UpdateLocation for event with id ' . $command->getItemId());
+            } catch (Exception $exception) {
+                $exceptions[] = sprintf('Event with id: %s caused an exception: %s', $event->getId(), $exception->getMessage());
```

Comment:
I would prefer to write the exception when it occurs so it is possible to abort the command when running it supervised.